MLS.Agent.Tests/CommandLine/PublishCommandTests.cs

```diff
@@ -0,0 +1,70 @@
+using System.CommandLine.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Options;
+using MLS.Agent.CommandLine;
+using MLS.Agent.Tools;
+using MLS.Agent.Tools.Tests;
+using WorkspaceServer.Tests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MLS.Agent.Tests.CommandLine
+{
+    public class PublishCommandTests
+    {
+        private const string CsprojContents = @"<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+</Project>
+";
+
+        public class WithMarkdownOutputFormat
+        {
+            private static PublishOptions Options(IDirectoryAccessor source, IDirectoryAccessor target = null) => new PublishOptions(source, target ?? source, PublishFormat.Markdown);
+
+            private readonly ITestOutputHelper _output;
+
+            public WithMarkdownOutputFormat(ITestOutputHelper output) => _output = output;
+
+            [Theory]
+            [InlineData("##Title")]
+            [InlineData("markdown with line \r\n break")]
+            [InlineData("markdown with linux line \n break")]
+            [InlineData("[link](https://try.dot.net/)")]
+            public async Task When_there_are_no_code_fence_annotations_markdown_is_unchanged(string markdown)
+            {
+                var rootDirectory = Create.EmptyWorkspace(isRebuildablePackage: true).Directory;
+
+                var directoryAccessor = new InMemoryDirectoryAccessor(rootDirectory)
+                {
+                    ("doc.md", markdown)
+                }.CreateFiles();
+
+
+                var targetDirectory = await DoPublish(directoryAccessor);
+
+                var files = targetDirectory.GetAllFilesRecursively().ToList();
+            }
+
+            private async Task<InMemoryDirectoryAccessor> DoPublish(IDirectoryAccessor directoryAccessor, IDirectoryAccessor target = null)
```

Comment:
It would help to clarify the purpose of the two `IDirectoryAccessor` parameters. Would `source` and `target` would make sense?

And did you mean to use `target` rather than initializing a new `targetDirectory` below?